src/lib/utils.ts

```diff
@@ -0,0 +1,40 @@
+export function deepFreeze<T extends object>(target: T): T {
+  Object.freeze(target);
+
+  const targetIsFunction = isFunction(target);
+
+  for (const prop of Object.getOwnPropertyNames(target)) {
+    // Do not freeze Ivy properties (e.g. router state)
+    // https://github.com/ngrx/platform/issues/2109#issuecomment-582689060
+    if (prop.startsWith('ɵ')) {
+      continue;
+    }
+
+    if (
+      Object.hasOwn(target, prop) &&
+      (targetIsFunction
+        ? prop !== 'caller' && prop !== 'callee' && prop !== 'arguments'
+        : true)
+    ) {
+      // @ts-ignore
+      const propValue = target[prop];
```

Comment:
will `(target as any)[prop]` help here to get rid of the `// @ts-ignore`? 👀

Reply:
I went with a Record<string, unknown>. What is better in that case?